lib/spack/spack/util/environment.py

```diff
@@ -24,6 +24,32 @@
 ##############################################################################
 import os
 
+system_paths = ['/', '/usr/', '/usr/local']
+suffixes = ['lib', 'lib64', 'include']
+system_dirs = sum(([os.path.join(p, s)
+                    for s in suffixes] for p in system_paths), [])
+system_bins = [os.path.join(p, 'bin') for p in system_paths]
+
+
+def filter_system_paths(paths):
+    return [p for p in paths if not any(p in s for s in system_dirs)]
```

Comment:
It seems to me that you are trying exact match here. If I am not wrong then:
```python
[p for p in paths if p not in system_dirs]
```
should do the same thing.